source/tutorials/nix-language.md

```diff
@@ -905,6 +905,10 @@ Example:
 
     /current
 
+:::{note}
+Before you start combining file system paths and antiquotation, you probably want to read about the [path-related impurities below](path-impurities).
```

Comment:
@jherland I'd like to keep the structure linear. Was there something specific that putting impure path behavior in front would have helped you with? Also, for reading the language we don't really care how things work once you use them, it's really just to understand what things are.

Reply:
Understood. I only stumbled into this by getting ahead of myself: I read the tutorial with a `nix repl` session open side-by-side, and after looking at antiquotation and then paths, I got curious about the difference between paths and strings. Specifically I remember thinking "so are paths and strings the same (like in Python 2), or are they separate (like with pathlib in Python 3)?" and next I tried to "string-ify" a path with antiquotation and all hell broke lose: 😉 

```nix
nix-repl> /dev/null
/dev/null

nix-repl> "${/dev/null}"
error (ignored): error: end of string reached
error: file '/dev/null' has an unsupported type

nix-repl> ./.
/home/jherland/nix-language-tutorial

nix-repl> "${./.}"
"/nix/store/0qg0178kca2qgdk83lz7f8zcjlw3sz95-nix-language-tutorial"

nix-repl> ../.
/home/jherland

nix-repl> "${../.}"
<HANGS AND MUST BE SIGKILLED!>
```

Most of this is explained later in the impurities section (incl. my proposed changes). But especially the last part was not a nice experience, and one that I can quickly see someone else running into as well.

I'm not sure what could be done in the tutorial to prevent this. As you say, the tutorial is about the language, and not really about the behavior, but we also do not want to discourage experimentation in `nix repl`...

Reply:
Okay, I agree this is a pitfall. I'll make a proposal to reword to carry less of your specific context and be more generally informative.